emcc.py

```diff
@@ -2128,7 +2135,11 @@ def compile_source_file(i, input_file):
         logger.debug('compiling source file: ' + input_file)
         output_file = get_object_filename(input_file)
         temp_files.append((i, output_file))
-        cmd = get_clang_command([input_file]) + ['-c', '-o', output_file]
+        if get_file_suffix(input_file) in ASSEMBLY_ENDINGS:
+          cmd = get_clang_command_asm([input_file])
+        else:
+          cmd = get_clang_command([input_file])
```

Comment:
how about putting this if inside `get_clang_command`? I think that would be cleaner.

alternatively, `get_clang_command` could be `get_clang_command_c` (to better compare with `_asm`), but I'd prefer that less personally.

Reply:
The previous version of this change did just that but it made get_clang_command much more complex.   See the last patch I uploaded for what it looked like before.

Reply:
https://github.com/emscripten-core/emscripten/pull/10975/commits/ae2aa1a44950132cf7fbb9399332282aeebd9a44

Reply:
I see, thanks. lgtm then